lib/spack/docs/build_systems/pythonpackage.rst

```diff
@@ -0,0 +1,733 @@
+``setup.py`` to the package like so:
+
+.. code-block:: python
+
+   build_directory = 'source'
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Alternate names for setup.py
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As previously mentioned, packages need to call their setup script ``setup.py``
+in order to be compatible with package managers like ``pip``. However, some
+packages like ``py-meep`` and ``py-adios`` come with multiple setup scripts,
+one for a serial build and another for a parallel build. You can override the
+default name to use like so:
+
+.. code-block:: python
+
+   def setup_file(self):
+       return 'setup-mpi.py' if '+mpi' in self.spec else 'setup.py'
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+PythonPackage vs. packages that use Python
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are many packages that make use of Python, but packages that depend
+on Python are not necessarily PythonPackages.
+
+"""""""""""""""""""""""
+Choosing a build system
+"""""""""""""""""""""""
+
+First of all, you need to select a build system. ``spack create`` usually
+does this for you, but if for whatever reason you need to do this manually,
+choose PythonPackage if and only if the package contains a ``setup.py``
+file.
+
+"""""""""""""""""""""""
+Choosing a package name
+"""""""""""""""""""""""
+
+Selecting the appropriate package name is a little more complicated
+than choosing the build system. By default, ``spack create`` will
+prepend ``py-`` to the beginning of the package name if it detects
+that the package uses the PythonPackage build system. However, there
+are occasionally packages that use PythonPackage that shouldn't
+start with ``py-``. For example:
+
+* busco
+* easybuild
+* httpie
+* mercurial
+* scons
+* snakemake
+
+The thing these packages have in common is that they are command-line
+tools that just so happen to be written in Python. Someone who wants
+to install ``mercurial`` with Spack isn't going to realize that it is
+written in Python, and they certainly aren't going to assume the package
+is called ``py-mercurial``. For this reason, we manually renamed the
+package to ``mercurial``.
+
+Likewise, there are occasionally packages that don't use the
+PythonPackage build system but should still be prepended with ``py-``.
+For example:
+
+* py-genders
+* py-py2cairo
+* py-pygobject
+* py-pygtk
+* py-pyqt
+* py-pyserial
+* py-sip
+* py-xpyb
+
+These packages are primarily used as Python libraries, not as
+command-line tools. You may see C/C++ packages that have optional
+Python language-bindings, such as:
+
+* antlr
+* cantera
+* conduit
+* pagmo
+* vtk
+
+Don't prepend these kind of packages with ``py-``. When in doubt,
+think about how this package will be used. Is it primarily a Python
+library that will be imported in other Python scripts? Or is it a
+command-line tool, or C/C++/Fortran program with optional Python
+modules? The former should be prepended with ``py-``, while the
+latter should not.
```

Comment:
Please address the issue.... if the upstream package is called `PyXyz`, should you call your package `py-xyz` or `py-pyxyz`?

Reply:
It should be called `py-pyxyz`. For example:
```console
$ spack list py-py
==> 67 packages.
py-py          py-pybind11         py-pydispatcher  py-pygobject   py-pypar        py-pyside          py-pytest-runner       py-pyutilib
py-py2bit      py-pybtex           py-pydot         py-pygpu       py-pyparsing    py-pysocks         py-pytest-xdist        py-pyvcf
py-py2cairo    py-pybtex-docutils  py-pyelftools    py-pygtk       py-pypeflow     py-pysqlite        py-python-daemon       py-pywavelets
py-py2neo      py-pychecker        py-pyfaidx       py-pylint      py-pyprof2html  py-pytables        py-python-engineio     py-pyyaml
py-py4j        py-pycodestyle      py-pyfasta       py-pymatgen    py-pyqt         py-pytest          py-python-gitlab
py-pyani       py-pycparser        py-pyfftw        py-pyminifier  py-pyrad        py-pytest-cov      py-python-levenshtein
py-pyasn1      py-pycrypto         py-pyflakes      py-pymol       py-pysam        py-pytest-flake8   py-python-socketio
py-pybedtools  py-pycurl           py-pygdbmi       py-pympler     py-pyscaf       py-pytest-httpbin  py-pythonqwt
py-pybigwig    py-pydatalog        py-pygments      py-pynn        py-pyserial     py-pytest-mock     py-pytz
```

Reply:
> spack create handles this anyway, so it shouldn't really be a problem.

Not everybody uses `spack create`.

Reply:
No need to manually change it, you can run `spack create --template python --force url` to change it.

P.S. If you ever think it isn't correctly identifying a package build system let me know and I can take a look.